tests/vmi_multus_test.go

```diff
@@ -1010,59 +1010,70 @@ func cidrToIP(cidr string) string {
+func configInterface(vmi *v1.VirtualMachineInstance, interfaceName, interfaceAddress string, userModifierPrefix ...string) error {
 	setStaticIpCmd := fmt.Sprintf("%sip addr add %s dev %s\n", strings.Join(userModifierPrefix, " "), interfaceAddress, interfaceName)
-	err := console.SafeExpectBatch(vmi, []expect.Batcher{
-		&expect.BSnd{S: "\n"},
-		&expect.BExp{R: console.PromptExpression},
-		&expect.BSnd{S: setStaticIpCmd},
-		&expect.BExp{R: console.PromptExpression},
-		&expect.BSnd{S: "echo $?\n"},
-		&expect.BExp{R: console.RetValue("0")},
-	}, 15)
-	Expect(err).ToNot(HaveOccurred(), "Failed to configure address %s for interface %s on VMI %s", interfaceAddress, interfaceName, vmi.Name)
+	err := runSafeCommand(vmi, setStaticIpCmd)
 
-	setIfaceUpCmd := fmt.Sprintf("ip link set %s up\n", interfaceName)
-	err = console.SafeExpectBatch(vmi, []expect.Batcher{
-		&expect.BSnd{S: "\n"},
-		&expect.BExp{R: console.PromptExpression},
-		&expect.BSnd{S: setIfaceUpCmd},
-		&expect.BExp{R: console.PromptExpression},
-		&expect.BSnd{S: "echo $?\n"},
-		&expect.BExp{R: console.RetValue("0")},
-	}, 15)
-	Expect(err).ToNot(HaveOccurred(), "Failed to set interface %s up on VMI %s", interfaceName, vmi.Name)
+	if err != nil {
+		return fmt.Errorf("could not configure address %s for interface %s on VMI %s: %w", interfaceAddress, interfaceName, vmi.Name, err)
+	}
+
+	return setInterfaceUp(vmi, interfaceName)
+}
+
+func checkInterface(vmi *v1.VirtualMachineInstance, interfaceName string) error {
+	cmdCheck := fmt.Sprintf("ip link show %s\n", interfaceName)
+	err := runSafeCommand(vmi, cmdCheck)
+
+	if err != nil {
+		return fmt.Errorf("could not check interface: interface %s was not found in the VMI %s: %w", interfaceName, vmi.Name, err)
+	}
+
+	return nil
 }
 
-func checkInterface(vmi *v1.VirtualMachineInstance, interfaceName string) {
+func checkMacAddress(vmi *v1.VirtualMachineInstance, interfaceName, macAddress string) error {
 	cmdCheck := fmt.Sprintf("ip link show %s\n", interfaceName)
 	err := console.SafeExpectBatch(vmi, []expect.Batcher{
 		&expect.BSnd{S: "\n"},
 		&expect.BExp{R: console.PromptExpression},
 		&expect.BSnd{S: cmdCheck},
-		&expect.BExp{R: console.PromptExpression},
+		&expect.BExp{R: macAddress},
 		&expect.BSnd{S: "echo $?\n"},
 		&expect.BExp{R: console.RetValue("0")},
 	}, 15)
-	Expect(err).ToNot(HaveOccurred(), "Interface %q was not found in the VMI %s within the given timeout", interfaceName, vmi.Name)
+
+	if err != nil {
+		return fmt.Errorf("could not check mac address of interface %s: MAC %s was not found in the VMI %s: %w", interfaceName, macAddress, vmi.Name, err)
+	}
+
+	return nil
 }
 
-func checkMacAddress(vmi *v1.VirtualMachineInstance, interfaceName, macAddress string) {
-	cmdCheck := fmt.Sprintf("ip link show %s\n", interfaceName)
-	err := console.SafeExpectBatch(vmi, []expect.Batcher{
+func setInterfaceUp(vmi *v1.VirtualMachineInstance, interfaceName string) error {
```

Comment:
just one place. I thought it will be more clear to read it this way. another option is to add `runSafeCommand("ip link set %s up\n")`  to
`configInterface` and assert on the error returning. I don't mind.